### PR TITLE
added CartService and CartController

### PR DESCRIPTION
**Pull request by aliaksharbachou**

### CODE REVIEW

File: src/main/java/internship/mbicycle/storify/controller/CartController.java [hunk 1]
```diff
@@ -1,31 +1,57 @@
...
     @GetMapping()
-    public List<Product> getCart(Principal principal) {
-        StorifyUser storifyUser = userService.getUserByEmail(principal.getName());
-        return storifyUser.getProfile().getCart().getProductList();
+    public CartDTO getCart(Principal principal) {
+        Cart cart = cartService.getCartByPrincipal(principal);
+        return cartConverter.convertCartToCartDTO(cart);
+    }
+
+    @DeleteMapping("/delete/{id}")
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/CartController.java:26]: "/delete" part is redundant

File: src/main/java/internship/mbicycle/storify/controller/CartController.java [hunk 2]
```diff
@@ -1,31 +1,57 @@
...
+    public CartDTO deleteProduct(@PathVariable(name = "id") long productId,
+                                 Principal principal) {
+        Cart cart = cartService.getCartByPrincipal(principal);
+        cartService.deleteProduct(cart, productId);
+        return cartConverter.convertCartToCartDTO(cart);
+    }
+
+    @PutMapping("/deleteByCount/{id}/{count}")
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/CartController.java:34]: "/deleteByCount" part is redundant

File: src/main/java/internship/mbicycle/storify/controller/CartController.java [hunk 3]
```diff
@@ -1,31 +1,57 @@
...
+    public CartDTO deleteProductByCount(@PathVariable(name = "id") long productId,
+                                        @PathVariable(name = "count") int count,
+                                        Principal principal) {
+        Cart cart = cartService.getCartByPrincipal(principal);
+        cartService.changeProductByCount(cart, productId, count);
+        return cartConverter.convertCartToCartDTO(cart);
+    }
+
+    @DeleteMapping("/delete-all")
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/CartController.java:43]: "/delete-all" is redundant

File: src/main/java/internship/mbicycle/storify/controller/CartController.java [hunk 4]
```diff
@@ -1,31 +1,57 @@
...
+    public CartDTO deleteAllProduct(Principal principal) {
+        Cart cart = cartService.getCartByPrincipal(principal);
+        cartService.deleteAllProduct(cart);
+        return cartConverter.convertCartToCartDTO(cart);
+    }
+
+    @PostMapping("/addProducts/{id}/{count}")
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/CartController.java:50]: "/addProducts" is redundant

File: src/main/java/internship/mbicycle/storify/converter/CartConverter.java [hunk 5]
```diff
@@ -2,43 +2,39 @@
...
+        for (Map.Entry<Long, Integer> entry : cart.getProductsMap().entrySet()) {
+            ProductDTO productDTO = productService.getProductById(entry.getKey());
+            ProductDetailInCartDTO productDetailInCartDTO = ProductDetailInCartDTO.builder()
+                    .productId(productDTO.getId())
+                    .product_name(productDTO.getProductName())
+                    .price(productDTO.getPrice())
+                    .amount(entry.getValue())
+                    .build();
+            productDetailInCartDTOList.add(productDetailInCartDTO);
+            sum += productDTO.getPrice() * entry.getValue();
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/converter/CartConverter.java:33]: try to use ObjectMapper instead

File: src/main/java/internship/mbicycle/storify/dto/ProductDetailInCartDTO.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class ProductDetailInCartDTO {
+
+    private Long productId;
+    private String product_name;
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/dto/ProductDetailInCartDTO.java:15]: product_name? What the hell with naming?)

**Review by Estrin-MBicycle — CHANGES_REQUESTED** [review]

File: src/main/java/internship/mbicycle/storify/converter/CartConverter.java (hunk 5 above)
Comment (OlgaEvdokimova) [inline src/main/java/internship/mbicycle/storify/converter/CartConverter.java:33]: try to use stream api

**Review by OlgaEvdokimova — CHANGES_REQUESTED** [review]

File: src/main/java/internship/mbicycle/storify/controller/CartController.java (hunk 3 above)
Comment (aliaksharbachou) [reply src/main/java/internship/mbicycle/storify/controller/CartController.java]: done

File: src/main/java/internship/mbicycle/storify/dto/ProductDetailInCartDTO.java (hunk 6 above)
Comment (aliaksharbachou) [reply src/main/java/internship/mbicycle/storify/dto/ProductDetailInCartDTO.java]: done

File: src/main/java/internship/mbicycle/storify/converter/CartConverter.java (hunk 5 above)
Comment (aliaksharbachou) [reply src/main/java/internship/mbicycle/storify/converter/CartConverter.java]: maybe there are examples of an object mapperwith a map?

Comment (aliaksharbachou) [reply src/main/java/internship/mbicycle/storify/converter/CartConverter.java]: done

File: src/main/java/internship/mbicycle/storify/controller/CartController.java (hunk 2 above)
Comment (aliaksharbachou) [reply src/main/java/internship/mbicycle/storify/controller/CartController.java]: done

File: src/main/java/internship/mbicycle/storify/converter/CartConverter.java (hunk 5 above)
Comment (aliaksharbachou) [reply src/main/java/internship/mbicycle/storify/converter/CartConverter.java]: done

**Pull request merged**